test/test-art-template.js

```diff
@@ -43,6 +43,44 @@ test('reply.view with art-template engine and custom templates folder', t => {
   })
 })
 
+test('reply.view for art-template without data-parameter', t => {
+  t.plan(6)
+  const fastify = Fastify()
+  const art = require('art-template')
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      'art-template': art
+    },
+    defaultContext: data,
```

Comment:
is the test passing also without `defaultContext`?

Reply:
I guess not - i copied the introducing test (which uses data-depending templates) and just modified the call of the `view`-function. Since i didn't touch the templating in general i expected it would be enough to just change the call-signature.

Reply:
:thinking: - i can make the test more atomic if that's what you're after.

Reply:
Looking at the code I think it should run since the is a fallback here
https://github.com/fastify/point-of-view/blob/39d8a4a1a4cbfe3728110fe29d4717ddd6ed61eb/index.js#L29

But I think it is more right to test also the case: "I didn't set the defaultContext and I don't want use the data parameter"

Reply:
Spent some time thinking about the test. I would really prefer to create a unit test for that.

The plugin is - in my opinion - handling errors at least arguable. Currently it returns an error-response by calling `this.send(err)` which doesn't allow content-generation and exposes errors to the public (which might be a security flaw).

I can of course write a test checking for the Error-response but with #134 in mind, that behaviour should probable be changed to throw an Error instead and or log something useful.

Reply:
> I would really prefer to create a unit test for that

I'm ok for add other tests: two is better than one 😄